drs_downloader_to_getm.wdl

```diff
@@ -43,61 +52,185 @@ task download {
+        cpu: select_first([cpu, "4"])
+        memory: select_first([memory,"16"]) + " GB"
+        disks: "local-disk " + select_first([disk, "128"]) + " HDD"
+        bootDiskSizeGb: select_first([boot_disk,"30"])
+    }
+}
 
-        # TODO Configure shared memory appropriately, if needed.
+task download {
+    meta {
+        description: "This task tests downloading DRS URIs using the downloader of your choice (getm, curl, wget, gsutil)."
+    }
+    parameter_meta {
+        manifest: "Manifest mapping DRS URIs to http/gs/s3 schema appropriate for use with getm."
+        downloader: "The downloader to use: (getm, curl, wget, gsutil)"
+        cpu: "runtime parameter - number of CPUs"
+        memory: "runtime parameter - amount of memory to allocate in GB. Default is: 16"
+        boot_disk: "runtime parameter - amount of boot disk space to allocate in GB. Default is: 50"
+        disk: "runtime parameter - amount of disk space to allocate in GB. Default is: 128"
+    }
+    input {
+        File manifest
+        String downloader
+        Int? cpu
+        Int? memory
+        Int? boot_disk
+        Int? disk
+    }
+    command <<<
+        set -eux o pipefail
 
-        # Debug: Output the lists of image_files
-        echo drs_uris: "~{sep='", "' drs_uris}"
+        CURRENT_DIR=$(pwd)
 
-        # Create a getm manifest for the DRS URIs
-        wget https://raw.githubusercontent.com/mbaumann-broad/getm-tests/dev/scripts/create_getm_manifest.py
-        python3 ./create_getm_manifest.py ~{getm_manifest_filename} "~{sep='" "' drs_uris}"
+        # Where all non-getm downloads go (wget, gsutil, and curl)
+        TMP_DL_DIR=/cromwell_root/speedtest3crdws3s
+        mkdir -p ${TMP_DL_DIR}
+
+        # Identify the runtime environment; Check Linux version
+        cat /etc/issue
+        uname -a
+        apt-get update
+        apt-get install -yq --no-install-recommends apt-utils git jq
 
-        # Download the files in the manifest
-        start_time=`date +%s`
-        time getm -c -v --manifest ~{getm_manifest_filename}
-        getm_exit_status=$?
-        echo "Getm exit status: "$getm_exit_status
-        end_time=`date +%s`
+        # Check available shared memory and disk usage before downloading
+        # TODO Configure shared memory appropriately, if needed.
+        df -h
 
-        # TODO Iterate over the manifest listing each file
-        ls -lR /cromwell_root/
+        if [ "~{downloader}" = "getm" ]; then
+            # Google project to bill for the gsutil downloads
+            GOOGLE_REQUESTER_PAYS_PROJECT=platform-dev-178517
+
+            apt-get install -yq --no-install-recommends virtualenv python3.8-dev
+            apt-get -yq --no-install-recommends install python3-pip
+            # Use a virtualenv
+            virtualenv -p python3.8 v38nv
+            set +u
+            . v38nv/bin/activate
+            set -u
+            # Check that we're really using python3.8
+            python --version
```

Comment:
I added this because it's just a good isolation habit, though it's not strictly necessary.  The way that this was initially set up, it was using two version of python, and this makes that obvious if using the standard methods.

Reply:
Python virtual environments do not generally provide additional isolation for containerized execution, although I do agree it’s nice to consistently use the same python executable. But a virtualenv is not needed for that.

In this case it is preferable to test the container as-is, or to at least match the environment of the Cromwell localizer. We would like to test in a typical user environment.

Reply:
PR for this: https://github.com/mbaumann-broad/getm-tests/pull/3